pkg/k8s/allowlist_test.go

```diff
@@ -15,35 +15,86 @@
 package k8s
 
 import (
+	"fmt"
 	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/crd/crdloader"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-func TestIsValueInAllowlist(t *testing.T) {
+func TestSupportsStateIntoSpecMerge(t *testing.T) {
 	tests := []struct {
 		name           string
-		kind           string
+		gvk            schema.GroupVersionKind
 		expectedResult bool
 	}{
 		{
-			name:           "ComputeInstance should support 'state-into-spec: merge'",
-			kind:           "ComputeInstance",
+			name: "ComputeInstance should support 'state-into-spec: merge'",
+			gvk: schema.GroupVersionKind{
+				Group:   "compute.cnrm.cloud.google.com",
+				Version: "v1beta1",
+				Kind:    "ComputeInstance",
+			},
 			expectedResult: true,
 		},
 		{
 			name: "AccessContextManagerServicePerimeterResource should not " +
 				"support 'state-into-spec: merge'",
-			kind:           "AccessContextManagerServicePerimeterResource",
+			gvk: schema.GroupVersionKind{
+				Group:   "accesscontextmanager.cnrm.cloud.google.com",
+				Version: "v1beta1",
+				Kind:    "AccessContextManagerServicePerimeterResource",
+			},
 			expectedResult: false,
 		},
 	}
 
 	for _, tc := range tests {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
-			actualResult := supportsStateIntoSpecMergeInKind(tc.kind)
+			actualResult := supportsStateIntoSpecMerge(tc.gvk)
 			if actualResult != tc.expectedResult {
 				t.Fatalf("got %v, want %v", actualResult, tc.expectedResult)
 			}
 		})
 	}
 }
+
+func TestOutputOnlyFieldsAreUnderObservedState(t *testing.T) {
+	crds, err := crdloader.LoadCRDs()
+	if err != nil {
+		t.Fatalf("error loading crds: %v", err)
+	}
+	for _, crd := range crds {
+		for _, version := range crd.Spec.Versions {
+			gvk := schema.GroupVersionKind{
+				Group:   crd.Spec.Group,
+				Version: version.Name,
+				Kind:    crd.Spec.Names.Kind,
+			}
+			t.Run(fmt.Sprintf("%s/%s/%s", gvk.Group, gvk.Version, gvk.Kind), func(t *testing.T) {
+				openAPISchema := version.Schema.OpenAPIV3Schema
+				prop := findOpenAPIProperty(openAPISchema, "status", "observedState")
+				hasObservedState := prop != nil
+				got := OutputOnlyFieldsAreUnderObservedState(gvk)
+
+				if hasObservedState != got {
```

Comment:
The test case fails because observedFields configuration needs to be taken into consideration as well: e.g. https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/config/servicemappings/compute.yaml#L36 .